client/src/components/InteractiveTools/InteractiveTools.test.js

```diff
@@ -1,58 +1,59 @@
+import { createTestingPinia } from "@pinia/testing";
+import { setActivePinia, PiniaVuePlugin } from "pinia";
 import InteractiveTools from "./InteractiveTools";
 import { mount } from "@vue/test-utils";
 import { getLocalVue } from "jest/helpers";
-import flushPromises from "flush-promises";
 import testInteractiveToolsResponse from "./testData/testInteractiveToolsResponse";
-import { createPinia } from "pinia";
-import MockAdapter from "axios-mock-adapter";
-import axios from "axios";
 
 describe("InteractiveTools/InteractiveTools.vue", () => {
     const localVue = getLocalVue();
-    const pinia = createPinia();
+    localVue.use(PiniaVuePlugin);
     let wrapper;
-    let axiosMock;
+    let testPinia;
 
     beforeEach(async () => {
-        axiosMock = new MockAdapter(axios);
-        axiosMock.onGet("/api/entry_points", { params: { running: true } }).reply(200, testInteractiveToolsResponse);
-        axiosMock.onDelete(new RegExp("/api/entry_points/*")).reply(200, { status: "ok", message: "ok" });
+        testPinia = createTestingPinia({
+            initialState: {
+                entryPointStore: {
+                    entryPoints: testInteractiveToolsResponse,
+                },
+            },
+        });
+        setActivePinia(testPinia);
         wrapper = mount(InteractiveTools, {
             localVue,
-            pinia,
+            pinia: testPinia,
         });
-        await flushPromises();
-    });
-
-    afterEach(() => {
-        axiosMock.restore();
     });
 
-    it("Interactive Tool Table renders", async () => {
-        const table = wrapper.find("#interactive-tool-table");
-        expect(table.exists() === true).toBeTruthy();
+    it("renders table with interactive tools", async () => {
+        expect(wrapper.get("#interactive-tool-table").exists() === true).toBeTruthy();
+        expect(wrapper.findAll("td > a").length).toBe(2);
     });
 
-    it("Interactive Tool should disappear after stop button pressed", async () => {
+    it("removes the interactive tool after stop button is pressed", async () => {
         function checkIfExists(tag, toolId) {
             return wrapper.find(tag + toolId).exists();
         }
+
         const toolId = testInteractiveToolsResponse[0].id;
         const tool = wrapper.vm.activeInteractiveTools.find((tool) => tool.id === toolId);
         expect(checkIfExists("#link-", toolId) === true).toBeTruthy();
         expect(tool.marked === undefined || false).toBeTruthy();
 
-        const checkbox = wrapper.find("#checkbox-" + toolId);
+        const checkbox = wrapper.get("#checkbox-" + toolId);
         checkbox.setChecked();
-        await flushPromises();
+        await new Promise((_) => setTimeout(_, 10));
```

Comment:
Try with:
```
        await checkbox.setChecked();
```

Reply:
Ohh I see... Not sure if this would help, but should the actual change in `activeInteractiveTools` come through the mocked-up Pinia store instead of altering the value directly in the wrapper.vm? It will probably not make any difference... but could be worth the try or maybe not, I'm afraid I'm not familiar yet with Pinia testing and mocking.

Remember you can use the `await` trick also with button.trigger(...) and other operations so you don't have to spam `flushPromises` after each call.